libr/core/cmd_anal.c

```diff
@@ -2035,19 +2036,83 @@ static int cmd_anal_fcn(RCore *core, const char *input) {
 			free (dbpath);
 			} break;
 		case'?':
-			r_core_cmd_help (core, help_afC);
+			r_core_cmd_help (core, help_afc);
 			break;
-		case 'l': //afCl list all function Calling conventions.
+		case 'l': //afcl list all function Calling conventions.
 			sdb_foreach (core->anal->sdb_cc, cc_print, NULL);
 			break;
 		case 'a':
 			eprintf ("Todo\n");
 			break;
+		case 'r': {	// afcr
+			int i;
+			char *out, *cmd, *regname, *tmp;
+			char *subvec_str = r_str_new ("");
+			char *json_str = r_str_new ("");
+			// if json_str initialize to NULL, it's possible for afcrj to output a (null)
+			// subvec_str and json_str should be valid until exiting this code block
+			bool json = input[3] == 'j'? true: false;
+			for (i = 0; i <= 11; i++) {
```

Comment:
emm, hah, because till now, the information stored in sdb (and all the calling convention we know) is:

For 32bits:
```
[0x00001f6b]> k anal/cc/*
borland=cc
cc.borland.arg1=eax
cc.borland.arg2=edx
cc.borland.arg3=ecx
cc.borland.argn=stack_rev
cc.borland.name=borland
cc.cdecl.argn=stack
cc.cdecl.name=cdecl
cc.cdecl.ret=eax
cc.fastcall.arg1=ecx
cc.fastcall.arg2=edx
cc.fastcall.argn=stack
cc.fastcall.name=fastcall
cc.fastcall.ret=eax
cc.optlink.arg1=eax
cc.optlink.arg2=edx
cc.optlink.arg3=ecx
cc.optlink.argn=stack
cc.optlink.name=optlink
cc.optlink.ret=eax
cc.pascal.argn=stack_rev
cc.pascal.name=pascal
cc.stdcall.argn=stack_rev
cc.stdcall.name=stdcall
cc.stdcall.ret=eax
cc.watcom.arg1=eax
cc.watcom.arg2=edx
cc.watcom.arg3=ebx
cc.watcom.arg4=ecx
cc.watcom.argn=stack
cc.watcom.name=watcom
cdecl=cc
default.cc=cdecl
fastcall=cc
optlink=cc
pascal=cc
stdcall=cc
watcom=cc
```

For 64bits:
```
[0x100000f60]> k anal/cc/*
amd64=cc
cc.amd64.arg1=rdi
cc.amd64.arg10=xmm3
cc.amd64.arg11=xmm4
cc.amd64.arg2=rsi
cc.amd64.arg3=rdx
cc.amd64.arg4=rcx
cc.amd64.arg5=r8
cc.amd64.arg6=r9
cc.amd64.arg7=xmm0
cc.amd64.arg8=xmm1
cc.amd64.arg9=xmm2
cc.amd64.name=amd64
cc.amd64.ret=rax
cc.ms.arg1=rcx
cc.ms.arg2=rdx
cc.ms.arg3=r9
cc.ms.argn=stack
cc.ms.name=ms
cc.ms.ret=rax
default.cc=amd64
ms=cc
```

11 is the largest number, and actually, xmm0..4 should not be normal args but float args, thus I made some check below.